plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClientModule.java

```diff
@@ -53,33 +54,75 @@ public void configure(Binder binder)
         configBinder(binder).bindConfig(OracleConfig.class);
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class)).setBinding().toInstance(ORACLE_MAX_LIST_EXPRESSIONS);
         newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
+        binder.bind(RetryingConnectionFactory.class).in(Scopes.SINGLETON);
     }
 
     @Provides
     @Singleton
     @ForBaseJdbc
-    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OracleConfig oracleConfig, OpenTelemetry openTelemetry)
-            throws SQLException
+    public ConnectionFactory connectionFactory(
+            OracleConfig oracleConfig,
+            OraclePoolConnectionFactory oraclePoolConnectionFactory,
+            RetryingConnectionFactory retryingConnectionFactory)
     {
-        Properties connectionProperties = new Properties();
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_INCLUDE_SYNONYMS, String.valueOf(oracleConfig.isSynonymsEnabled()));
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_REPORT_REMARKS, String.valueOf(oracleConfig.isRemarksReportingEnabled()));
-
         if (oracleConfig.isConnectionPoolEnabled()) {
-            return new OraclePoolConnectionFactory(
-                    config.getConnectionUrl(),
-                    connectionProperties,
-                    credentialProvider,
-                    oracleConfig.getConnectionPoolMinSize(),
-                    oracleConfig.getConnectionPoolMaxSize(),
-                    oracleConfig.getInactiveConnectionTimeout());
+            return oraclePoolConnectionFactory;
+        }
+        else {
+            return retryingConnectionFactory;
         }
+    }
 
-        return new RetryingConnectionFactory(new DriverConnectionFactory(
-                new OracleDriver(),
+    @Provides
+    @Singleton
+    public OraclePoolConnectionFactory getOraclePoolConnectionFactory(
```

Comment:
In case of pooling there is no retrying or open telemetry? Why?

Reply:
I was refactoring the existing code (ie. without changing it). So the question is I believe to the original author of this code? Can you help me identifying who is that?
![Uploading Screenshot 2023-10-02 at 11.03.51.png…]()

Reply:
I understand. I am not sure if we need to blame people. We could try to own things.

@hashhar @wendigo @ebyhr What do you think here?

Reply:
Was I blaming? If you felt that way, I'm sorry. That was not my intention. I just wanted to point out that I have no experience with this code, so cannot answer your question. Also wanted to point out that there are more valuable people WRT to answering that. But you already asked them, so I suppose we can move forward.

Reply:
I don't know. Can you investigate this @dominikzalewski since you're already working on this/touching that portion of code?

Reply:
I believe this is out of scope?

Acceptance criteria
- no new regressions
- unit tests for new framework that it actually retries
- it is possible to implement retry for any connector in future that fails sometimes to open a connection
- no custom retrying in any jdbc connector, all connectors use new framework

Reply:
> In case of pooling there is no retrying or open telemetry? Why?

I believe open telemetry was alway there? `openTelemetry` param is passed in both `OraclePoolConnectionFactory` constructor and `DriverConnectionFactory` constructor.

Now `RetryingConnectionFactory` is moved higher in the chain. It wraps anything that goes out of the `OracleClientModule` marked as `@ForBaseJdbc`. Let me know if it fixes it for you.

Reply:
yup, it was initially missing but was added in the meantime in https://github.com/trinodb/trino/pull/19241, so this comment is resolved